### DISCUSSION TITLE
array material

### DISCUSSION THREAD

**Discussion by mortezaaesmaeilpour** [idaholab/moose#21281] [Q&A General] [answered]
Dear all

Do you have any idea how can I define an array of material properties (like concentrations of different components)? As far as I know, material type can be real, vector, and rank2 tensor. 

Thanks

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can have a look at this one:
https://mooseframework.inl.gov/source/materials/GenericConstantArray.html

Guillaume

**Reply by mortezaaesmaeilpour** [reply]
Unfortunately, my material properties are not constant values. They are calculated through complicated functions of pressure, enthalpy, ...

**Reply by GiudGiud** [reply]
You ll have to use this as a template to make a new array material property that has the dependencies you need. 
We wont have the exact formula you need pre-implemented even for Real / Vector / RankTwoTensor materials.
